src/Tools/TestDiscoveryWorker/Program.cs

```diff
@@ -12,96 +12,103 @@
+            return context.LoadFromAssemblyPath(assemblyPath);
+        }
 
-            return null;
-        };
+        return null;
+    };
 #endif
 
-        string testDescriptor = Path.GetFileName(assemblyFileName);
+    string assemblyFileName = Path.GetFileName(assemblyFilePath);
 #if NET
-        testDescriptor += " (.NET Core)";
+    string tfm = "(.NET Core)";
 #else
-    testDescriptor += " (.NET Framework)";
+    string tfm = "(.NET Framework)";
 #endif
 
-        await Console.Out.WriteLineAsync($"Discovering tests in {testDescriptor}...").ConfigureAwait(false);
+    Console.Write($"Discovering tests in {tfm} {assemblyFileName} ... ");
 
-        using var xunit = new XunitFrontController(AppDomainSupport.IfAvailable, assemblyFileName, shadowCopy: false);
-        var configuration = ConfigReader.Load(assemblyFileName, configFileName: null);
-        var sink = new Sink();
-        xunit.Find(includeSourceInformation: false,
-                   messageSink: sink,
-                   discoveryOptions: TestFrameworkOptions.ForDiscovery(configuration));
+    using var xunit = new XunitFrontController(AppDomainSupport.IfAvailable, assemblyFilePath, shadowCopy: false);
+    var configuration = ConfigReader.Load(assemblyFileName, configFileName: null);
+    var sink = new Sink();
+    xunit.Find(includeSourceInformation: false,
+                messageSink: sink,
+                discoveryOptions: TestFrameworkOptions.ForDiscovery(configuration));
 
-        var testsToWrite = new HashSet<string>();
-        await foreach (var fullyQualifiedName in sink.GetTestCaseNamesAsync())
-        {
-            testsToWrite.Add(fullyQualifiedName);
-        }
-
-        if (sink.AnyWriteFailures)
-        {
-            await Console.Error.WriteLineAsync($"Channel failed to write for '{assemblyFileName}'").ConfigureAwait(false);
-            return ExitFailure;
-        }
-
-#if NET6_0_OR_GREATER
-        await Console.Out.WriteLineAsync($"Discovered {testsToWrite.Count} tests in {testDescriptor}").ConfigureAwait(false);
-#else
-        await Console.Out.WriteLineAsync($"Discovered {testsToWrite.Count} tests in {testDescriptor}").ConfigureAwait(false);
-#endif
+    var testsToWrite = new HashSet<string>();
+    await foreach (var fullyQualifiedName in sink.GetTestCaseNamesAsync())
+    {
+        testsToWrite.Add(fullyQualifiedName);
+    }
 
-        var directory = Path.GetDirectoryName(assemblyFileName);
-        using var fileStream = File.Create(Path.Combine(directory!, "testlist.json"));
-        await JsonSerializer.SerializeAsync(fileStream, testsToWrite).ConfigureAwait(false);
-        return ExitSuccess;
+    if (sink.AnyWriteFailures)
+    {
+        Console.WriteLine($"Channel failed to write for '{assemblyFileName}'");
+        return ExitFailure;
     }
 
+    Console.WriteLine($"{testsToWrite.Count} found");
+
+    using var fileStream = new FileStream(outputFilePath, FileMode.OpenOrCreate, FileAccess.Write, FileShare.None);
```

Comment:
If the file already exists, will this code do the right thing? I would expect existing file contents to get replaced with the new contents?